### ISSUE TITLE
Final Fantasy V (382890)

### ISSUE THREAD

**Issue by Zero86Sk** [ValveSoftware/Proton#447]
Fonts in the game are just a mess and unreadable.
Maybe some core fonts are missing?

Game seems to run fine otherwise

This is my system:
https://gist.github.com/Zero86Sk/d629f657c1ae861c36657cdf238a7765

**Comment by Zero86Sk** [comment]
Here is my log from steam
[steam-382890.log](https://github.com/ValveSoftware/Proton/files/2320131/steam-382890.log)

**Comment by TomasFitz** [comment]
Confirming this issue on Ubuntu 18.10, 4.18.0-11-generic kernel

**Comment by steven-a-wilson** [comment]
This was reported upstream before Proton was released. If I understand things correctly, the bug is that Wine's GDI+ can't draw without alpha blending, and GdipGraphicsClear() is implemented by drawing a rectangle of the specified color. FFV clears the Graphics objects that it uses to draw text with a 100% transparent color, which makes the "clear" operation do nothing.

https://bugs.winehq.org/show_bug.cgi?id=42585

**Comment by RyanEiri** [comment]
I have the same problem. 

https://gist.github.com/d7gonzo/5d7eee0740650bbe8cc0cc7ef880e64d

**Comment by kust2708** [comment]
> I have the same problem.
> 
> https://gist.github.com/d7gonzo/5d7eee0740650bbe8cc0cc7ef880e64d

You need to install gdiplus using protontricks (protontricks 382890 gdiplus) or with winetricks (WINEPREFIX=~/.local/share/Steam/steamapps/compatdata/382890/pfx winetricks gdiplus). It solve the issue on my side.

**Comment by RyanEiri** [comment]
> You need to install gdiplus using protontricks (protontricks 382890 gdiplus) or with winetricks (WINEPREFIX=~/.local/share/Steam/steamapps/compatdata/382890/pfx winetricks gdiplus). It solve the issue on my side.

protontricks is exactly the kind of tool I was hoping for. Makes it much easier to find the prefixes. Thanks for pointing it out  @kust2708! 

Easy to install with python3: [https://github.com/Matoking/protontricks](url)

**Comment by shoober420** [comment]
# Compatibility Report
- Name of the game with compatibility issues: FINAL FANTASY V
- Steam AppID of the game: https://steamdb.info/app/382890/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.59
- Kernel version: 5.5.2
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 4.11-12

LOG:
[steam-382892.zip](https://github.com/ValveSoftware/Proton/files/4173935/steam-382892.zip)

## Symptoms
Font is rendered incorrectly
![7616B95F-F3B0-47E6-B89B-8B1D9D6CADBC](https://user-images.githubusercontent.com/5150898/74081978-f9e37b80-4a22-11ea-9b05-d33a54ec952e.jpeg)

![0BD3BD9D-7D97-49B9-9B79-616E8F9F032C](https://user-images.githubusercontent.com/5150898/74081982-08319780-4a23-11ea-8377-120670f70311.jpeg)

![E7C9E946-7E14-4BEE-8444-F976FA85933D](https://user-images.githubusercontent.com/5150898/74081985-12539600-4a23-11ea-9890-5761631b9fca.jpeg)

**Comment by steven-a-wilson** [comment]
The upstream bug has been marked fixed and the fix is expected to be in Wine 5.4.

https://bugs.winehq.org/show_bug.cgi?id=42585#c10

**Comment by steven-a-wilson** [comment]
This appears to be fixed in Proton 5.13 (presumably by way of Wine 5.0.1).

**Comment by ugly95** [comment]
> This appears to be fixed in Proton 5.13 (presumably by way of Wine 5.0.1).

Yes, I can confirm that the overlapping text issue has been fixed.  The game appears to be playable out-of-the-box now.

The text still isn't perfect.  It appears to be a little bit pixellated.  Here is a screenshot using 5.13:
![Screenshot from 2020-11-16 01-58-02](https://user-images.githubusercontent.com/34800812/100556330-b4bd5e00-326f-11eb-9477-cadfcd9d5c4d.png)

For reference, this is from one of the official screenshots on the game's page on Steam:
![ss_dbc9e2a429fa118048fd32666c1d117737959687](https://user-images.githubusercontent.com/34800812/100556349-e6362980-326f-11eb-892e-2a75c930e92e.jpg)

You can see that the text in the official screenshot is blended, while the 5.13 screenshot has jagged edges.

**Comment by garatma** [comment]
I'm also having trouble with the fonts in this game.

I tried multiple versions of Proton, from 3.x to 6.x, with or without **gdiplus** and/or **corefonts** (both installed via protontricks). Every combination landed on the same result: _some_ characters just won't render. Here are some screenshots.

![mainmenu](https://user-images.githubusercontent.com/48687638/119185738-7f8b0f00-ba4d-11eb-888f-73c22fe338d2.png)
The main menu seems fine, nothing is missing.

![missing](https://user-images.githubusercontent.com/48687638/119185744-80bc3c00-ba4d-11eb-8ef0-05602c1be8f1.png)
Here you can really see that most characters aren't being rendered.

![nothing](https://user-images.githubusercontent.com/48687638/119185747-81ed6900-ba4d-11eb-981b-a652d09c5d7f.png)
And on this one nothing is being rendered.

My system information: https://gist.github.com/garatma/5e29d2673060a9390e5d93b57fe8c045

Also, following [this guide](https://steamcommunity.com/sharedfiles/filedetails/?id=2245082825), I changed the fonts that the game apparently uses and pointed them to fonts that I had installed. But this only changed the characters that were rendered, while the others were still missing.

Any ideas what might be going on? Am I missing some dependencies/fonts on my system? I tried installing and uninstalling some fonts, to no success. Thanks!

**Comment by ugly95** [comment]
I can confirm the unreadable text.

This must be a new issue.  

In my earlier post on 29 Nov 2020, I posted a few screenshots and the text was working.  Testing now in that same scene the text is missing.

Even going back to Proton 5.13, which I was using in my previous post, still has the issue.  Although, 5.13 seems to show more text for the in-game dialog than Proton 6.3.  But in both, the menu text is almost entirely missing.

Could something in newer version of Mesa be causing this?

**Comment by garatma** [comment]
> Could something in newer version of Mesa be causing this?

It is possible. The text isn't rendered properly on my Arch install with _mesa 21.1.0-1_ but I installed the game on Linux Mint, which has _mesa 20_, and the text is displayed correctly with Proton 6.3 and Proton 5.13, just like in the pictures you uploaded last year.

**Comment by RyanEiri** [comment]
I can confirm the issue with default Proton compatibility install. Even the title screen has blank spaces instead of text for this install.

Ubuntu 20.04
5.8 kernel
Mesa 21.0.3

![382890_screenshots_20210523124802_1](https://user-images.githubusercontent.com/3592569/119271195-20043f00-bbc6-11eb-906f-62365e5eb570.jpg)

**Comment by ugly95** [comment]
> > Could something in newer version of Mesa be causing this?
> 
> It is possible. The text isn't rendered properly on my Arch install with _mesa 21.1.0-1_ but I installed the game on Linux Mint, which has _mesa 20_, and the text is displayed correctly with Proton 6.3 and Proton 5.13, just like in the pictures you uploaded last year.

I'm on Linux Mint, but I am using the kisak-mesa PPA, so I'm on mesa 21.1.1.

I've been using that PPA for a while, so last year when I uploaded the screenshots I probably was probably on mesa 20.

**Comment by kasha695** [comment]
This issue has been fixed in Mesa in this [Commit 9b8bc712](https://gitlab.freedesktop.org/mesa/mesa/-/commit/9b8bc712b2b0f60eb090a03b43156cd98146917c)
Link to the bug report in Mesa as well: [Final Fantasy V (Old version): Random text characters are not displayed](https://gitlab.freedesktop.org/mesa/mesa/-/issues/5273)

**Comment by ugly95** [comment]
With the update to the kisak-mesa PPA today to Mesa 21.3.0 the bug fix is in and the game works properly again.
